File: archives/1924.py
import sys
input = sys.stdin.readline

import datetime
x, y = map(int, input().split())
day1, day2 = datetime.date(2007, 1, 1), datetime.date(2007, x, y)
num = (day2-day1).days % 7
if num == 0:
    print("MON")
elif num == 1:
    print("TUE")
elif num == 2:
    print("WED")
elif num == 3:
    print("THU")
elif num == 4:
    print("FRI")
elif num == 5:
    print("SAT")
else:
    print("SUN")

File: archives/cpp/1924.cpp
#include <bits/stdc++.h>
#define FOR(i, start, end) for (int i = start; i < end; i++)
#define endl "\n"
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int month[13] = {0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
    string week[7] = {"SUN", "MON", "TUE", "WED", "THU", "FRI", "SAT"};

    int x, y; cin >> x >> y;
    FOR(i, 1, x){
        y += month[i];
    }
    cout << week[y % 7];

    return 0;
}